### PR TITLE
#2854 Session (host) information is not reported with VMs

### PR DESCRIPTION
**Pull request by MikhailArkhipov**

### CODE REVIEW

File: src/Host/Client/Impl/Session/RSession.cs [hunk 1]
```diff
@@ -410,7 +410,11 @@ public Task StartHostAsync(RHostStartupInfo startupInfo, IRSessionCallback callb
 
                     var wd = startupInfo.WorkingDirectory;
                     if (!IsRemote && !string.IsNullOrEmpty(wd) && wd.HasReadPermissions()) {
-                        await evaluation.SetWorkingDirectoryAsync(wd);
+                        try {
+                            await evaluation.SetWorkingDirectoryAsync(wd);
+                        } catch(REvaluationException) {
+                            await evaluation.SetDefaultWorkingDirectoryAsync();
```
Comment (AlexanderSher) [inline src/Host/Client/Impl/Session/RSession.cs]: Is it really `REvaluationException`?

**Comment by MikhailArkhipov** [comment]
It was 

```
Error in setwd("C:/Users/alsher/Documents") : 
  cannot change working directory
```

I think it translates to REval. I added access check, but hardening it more

File: src/Host/Client/Impl/Session/RSession.cs (hunk 1 above)
Comment (MikhailArkhipov) [inline src/Host/Client/Impl/Session/RSession.cs]: `cannot change working directory` comes as REval?

**Comment by MikhailArkhipov** [comment]
OK, I need to build this along with earlier merge to verify SQL issue is fixed

File: src/Host/Client/Impl/Session/RSession.cs (hunk 1 above)
Comment (int19h) [inline src/Host/Client/Impl/Session/RSession.cs]: The only thing that can come from` EvaluateAsync` (which is what it ultimately does) is `REvaluationException` and `OperationCanceledException` - if there's something else, it's a bug in and of itself.

**Pull request merged**